app/utils/sweats_message_handler.js

```diff
@@ -0,0 +1,141 @@
+        const intervalId = listenForReactions(channelId, messageId);
+        if (intervalId === null) {
+            return;
+        }
+        setTimeout(() => clearInterval(intervalId), 1000 * 60 * 60 * 24); // 24 hours
+    } catch (error) {
+        console.error(error);
+        throw new Error('Error handling sweats message');
+    }
+}
+
+
+function listenForReactions(channelId, messageId) {
+    const endpoint = `channels/${channelId}/messages/${messageId}/reactions/%F0%9F%92%A7`;
+    const listener = new EventEmitter();
+    let previousUsers = [];
+    try {
+        const reactioncheck = setInterval(async () => {
+            var response;
+            try {
+            response = await DiscordRequest(endpoint, {
+                method: 'GET',
+                });
+            }catch (error) {
+                console.log(error);
+                const errorObj = JSON.parse(error.message);
+                if (errorObj.code === 10008) {
+                    clearInterval(reactioncheck);
+                    console.error('Message not found, stopping reaction check');
+                    return null;
+                } else {
+                    console.log(error);
+                    return reactioncheck;
+                }
+            }
+            const users = await response.json();
+            const userIds = users.map(user => user.id);
+
+            if (JSON.stringify(userIds) !== JSON.stringify(previousUsers)) {
+                previousUsers = userIds;
+                listener.emit('reaction', users);
+            }
+        }, 5000); // Check every 5 seconds
+        
+
+        listener.on('reaction', async (users) => {
+            users = users.filter(user => !user.bot);
+            if (users.length != 0) {
+                removeOwnReaction(channelId, messageId);
+            } else {
+                addOwnReaction(channelId, messageId);
+            }
+            editMessage(channelId, messageId, users);
+        });
+
+        return reactioncheck;
+    } catch (error) {
+        console.error(error);
+    }
+    return null;
+}
+
+
+export async function sendMessage(channelId) {
+    try {
+        const endpoint = `channels/${channelId}/messages`;
+        const response = await DiscordRequest(endpoint, {
+            method: 'POST',
+            body: {
+                embeds: [{
+                    title: 'Sweats Today!',
+                    description: 'React with 💧 to sign up for sweats today!\nCurrent signups: None',
+                    color: green,
+                }],
+            }
+        });
+        const message = await response.json();
+        addOwnReaction(channelId, message.id);
+        return message.id;
+    } catch (error) {
+        console.error(error);
+    }
+    return null;
+}
+
+export async function editMessage(channelId, messageId, users) {
+    var color = green;
+    if (users.length >= 5) {
+        color = red;
+    }
```

Comment:
(Suggestion: architecture) If taking the above advice, this can be moved out of this function and instead be a trigger for it (e.g. if a reaction is added and the users are > 5, call the edit function)